src/dbos-executor.ts

```diff
@@ -852,6 +853,25 @@ export class DBOSExecutor implements DBOSExecutorContext {
       }
     }
 
+    const handleWorkflowError = async (err: Error) => {
+      // Record the error.
+      const e = err as Error & { dbos_already_logged?: boolean };
+      this.logger.error(e);
+      e.dbos_already_logged = true;
+      if (wCtxt.isTempWorkflow) {
+        internalStatus.workflowName = `${DBOSExecutor.tempWorkflowName}-${wCtxt.tempWfOperationType}-${wCtxt.tempWfOperationName}`;
```

Comment:
I think we can remove this special case on temp workflows now. Previously, we don't run init workflow for temp workflows so we had to calculate the name here. But now we always first init the workflow, and the name is already there in the database.

Reply:
Let's do this cleanup in another PR?